step/step_test.go

```diff
@@ -0,0 +1,115 @@
+package step
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/bitrise-io/go-utils/v2/log"
+	"github.com/bitrise-steplib/steps-go-test/mocks"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGoTestRunner_Run_WhenTestSucceedItWritesCodeCoverageToFile(t *testing.T) {
+	// Create run options
+	packages := []string{"./..."}
+	outputDir := "output"
+	opts := RunOpts{
+		Packages:  packages,
+		OutputDir: outputDir,
+	}
+
+	// Expected result
+	wantCoveragePth := filepath.Join(outputDir, "go_code_coverage.txt")
+	wantRunResult := &RunResult{
+		CodeCoveragePth: wantCoveragePth,
+	}
+
+	// Create mocks
+	mockCmdFactory := new(mocks.Factory)
```

Comment:
You could use the generated NewFactory() method. It automatically runs AssertExpectations(t) on cleanup, so it is harder to forget.

Reply:
Nice one, thank you, updated in https://github.com/bitrise-steplib/steps-go-test/pull/13/commits/5cea7fe0183e0b99fc0be67f237317c21419e92d